src/SmoFacade/Listener.cs

```diff
@@ -128,11 +129,21 @@ private static string AgListenerName(string dataSource)
       return dotIndex >= 0 ? dataSource.Remove(dotIndex) : dataSource;
     }
 
-    private static Server AgListenerNameToServer(ref SqlConnectionStringBuilder connBuilder, string agInstanceName,
-      string credentailName)
+    private static Server AgListenerNameToServer(ref SqlConnectionStringBuilder connBuilder, string agInstanceName)
     {
-      connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
-      return new Server(connBuilder.ToString(), credentailName);
+      try {
+        connBuilder.DataSource = Dns.GetHostEntry(agInstanceName).HostName;
+      }
+      catch(Exception) {

```

Comment:
Is there a more specific exception we should catch here?

[Probably irrelevant, see below]

Reply:
SocketException
https://docs.microsoft.com/en-us/dotnet/api/system.net.dns.gethostname?view=netcore-3.1